internal/controller/apiextensions/claim/configurator.go

```diff
@@ -74,20 +85,20 @@ func Configure(_ context.Context, cm resource.CompositeClaim, cp resource.Compos
 		return nil
 	}
 
-	// TODO(negz): Consider nesting all user-specified spec values under a
-	// predictable object like spec.parameters so we can propagate _only_ user
-	// specified fields. Maintaining a set of keys to delete here seems error
-	// prone. Note that deleting these keys isn't always necessary; the CRD will
-	// drop them if preserveUnknownFields is false, but that is not the default
-	// for pre-v1 CRDs.
-	i, _ := fieldpath.Pave(ucm.Object).GetValue("spec")
-	spec, ok := i.(map[string]interface{})
+	icmSpec := ucm.Object["spec"]
+	spec, ok := icmSpec.(map[string]interface{})
 	if !ok {
 		return errors.New(errUnsupportedClaimSpec)
 	}
 
-	_ = fieldpath.Pave(ucp.Object).SetValue("spec", filter(spec, xcrd.FilterClaimSpecProps...))
+	// Delete base claim fields when configuring composite spec
+	baseClaimSpec := xcrd.CompositeResourceClaimSpecProps()
+	// keep composition selection fields
+	delete(baseClaimSpec, "compositionRef")
```

Comment:
It seemed slightly clearer here to filter all keys and then specifically not filter the two composition selection fields.

Reply:
This seems reasonable, but it did take me a few reads to wrap my head around. Perhaps a slightly more detailed comment before this logic would be useful? That said, I just tried writing one up and it didn't say much more than your commentary here already does.